UNIXFSv1.md

```diff
@@ -0,0 +1,487 @@
+
+	return l
+```
+
+This will tell you which offset inside this node the children at the corresponding index starts to cover.
+
+####### `PBNode.Links[].Name` with Files
+
+They SHOULD be absent from the protobuf message, however for historic reasons we allow empty file names.
+
+If this field is present and non empty the file is invalid and you MUST error.
+
+####### `Blocksize` of a dag-pb file
+
+This is not a field present in the block directly, but rather a computable property of a file which would be used in parent files.
+It is the sum of the length of the `Data.Data` field plus the sum of all link's blocksizes.
+
+####### `PBNode.Data.Filesize`
+
+If present, this field must be equal to the `Blocksize` computation above, else the file is invalid.
+For now, the usage of this field is unknown given this could be computed from other fields, but some implementations emit it.
+
+####### Path resolution
+
+Any attempt of path resolution on files MUST error.
+
+###### `Directory` Type
+
+A directory node is a named collection of file.
+
+The minimum valid `PBNode.Data` field for a directory is (pseudo-json): `{"Type":"Directory"}`, other values are covered in Metadata.
+
+Every link in the Links list is an entry (children) of the directory, and the `PBNode.Links[].Name` field give you the name.
+
+####### Link ordering
+
+The cannonical sorting order is lexicographical over the names.
+
+In theory there is no reason an encoder couldn't use an other ordering, however this lose some of it's meaning when mapped into most file systems today (most file systems consider directories are unordered-key-value objects).
+
+A decoder SHOULD if it can, preserve the order of the original files in however it consume thoses names.
+
+However when some implementation decode, modify then reencode some, the orignal links order fully lose it's meaning. (given that there is no way to indicate which sorting was used originally)
+
+####### Path Resolution
+
+Pop the left most component of the path, and try to match it to one of the Name in Links.
+
+<!--TODO: check Kubo does this-->If you find a match you can then remember the CID. You MUST continue your search, however if you find a match again you MUST error.
+
+Assuming no errors were raised, you can continue to the path resolution on the mainaing component and on the CID you poped.
+
+####### Duplicate names
+
+Duplicate names are not allowed, if two identical names are present in an directory, the decoder MUST error.
+
+##### `Symlink` type
+
+<!--TODO: check that this is true-->Symlinks MUST NOT have childs.
+
+Their Data.Data field is a path you may follow resolution at after appending it in front of the current path.
+
+###### Path resolution on symlinks
+
+Currently symlinks are not followable, that mean implementations needs to return symlinks objects and fail if a consumer tries to follow it through.
+
+This is a SHOULD level, you probably wont break much things if you start following them.
```

Comment:
It's unclear how to me how the unixfs format rather than the consumer would be able to follow symlinks, since the path doesn't provide a CID destination and the destination context will not be reliably clear at the point of link decoding (e.g. across mount, etc.)